src/main/java/org/junit/runners/parameterized/BlockJUnit4ClassRunnerWithParameters.java

```diff
@@ -73,6 +73,10 @@ private Object createTestUsingFieldInjection() throws Exception {
             int index = annotation.value();
             try {
                 field.set(testClassInstance, parameters[index]);
+            } catch (IllegalAccessException e) {
+                throw new Exception("Cannot set parameter '" + field.getName()
```

Comment:
You're right with the cause. The problem with creating a new `IllegalAccessException` is that it doesn't have a constructor with a `cause` argument.